File: leetcode_problems/cpp/lc1984_min_diff_between_high_and_low_k_scores.cpp
#include <vector>
#include <algorithm>
using std::vector;

class Solution
{
public:
    int minimumDifference(vector<int> &nums, int k)
    {
        int n = nums.size();
        std::sort(nums.begin(), nums.end());
        int max = 10000001;
        for (int i = 0; i <= n - k; i++)
        {
            max = std::min(max, nums[i + k - 1] - nums[i]);
        }
        return max;
    }
};

File: leetcode_problems/typescript/src/lc1984_min_diff_between_high_and_low_k_scores.ts
function minimumDifference(nums: number[], k: number): number {
  let n = nums.length;
  nums.sort((a, b) => a - b);
  let max = 10000001;
  for (let i = 0; i <= n - k; i++) {
    max = Math.min(max, nums[i + k - 1] - nums[i]);
  }
  return max;
}
